pandas/core/groupby/grouper.py

```diff
@@ -144,17 +144,18 @@ def _get_grouper(self, obj, validate=True):
         )
         return self.binner, self.grouper, self.obj
 
-    def _set_grouper(self, obj, sort=False):
+    def _set_grouper(self, obj: NDFrame, sort: bool = False):
```

Comment:
Return value can be annotated as None here right?

Reply:
Will double-check and add if so

Reply:
no, it returns self.grouper